projects/packages/explat/src/class-rest-controller.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * The ExPlat Rest Controller class.
+ * Registers the REST routes for ExPlat backend
+ *
+ * @package automattic/jetpack-explat
+ */
+
+namespace Automattic\Jetpack\Explat;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Jetpack_Connection;
+use WP_Error;
+use WP_REST_Server;
+
+/**
+ * Registers general REST routes for ExPlat.
+ */
+class REST_Controller {
+	/**
+	 * Namespace for the REST API.
+	 *
+	 * @var string
+	 */
+	public static $namespace = 'jetpack/v4/explat';
+
+	/**
+	 * Current version of the ExPlat API and components
+	 *
+	 * @var string
+	 */
+	const EXPLAT_API_VERSION = '0.1.0';
+
+	/**
+	 * Base API URI for WordPress.com
+	 *
+	 * @var string
+	 */
+	const WPCOM_API_BASE_URL = 'https://public-api.wordpress.com/wpcom/v2';
+
+	/**
+	 * Registers the REST routes.
+	 *
+	 * @access public
+	 * @static
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			static::$namespace,
+			'assignments',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_assignments' ),
+				'permission_callback' => '__return_true',
+				'args'                => array(
+					'experiment_name'   => array(
+						'type' => 'string',
+					),
+					'anon_id'           => array(
+						'type' => 'string',
+					),
+					'as_connected_user' => array(
+						'type' => 'boolean',
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get the assignments for a given experiment and anon_id
+	 *
+	 * @param WP_REST_Request $request The REST request object.
+	 * @return WP_REST_Response
+	 */
+	public function get_assignments( $request ) {
+		$response          = null;
+		$is_user_connected = ( new Jetpack_Connection() )->is_user_connected();
+		$request_path      = '/experiments/' . self::EXPLAT_API_VERSION . '/assignments/jetpack';
+		$args              = array(
+			'experiment_name' => $request['experiment_name'],
+			'anon_id'         => $request['anon_id'],
+		);
+
+		if ( $request['as_connected_user'] && $is_user_connected ) {
+			$response = Client::wpcom_json_api_request_as_user(
```

Comment:
Cool I didn't know this was a thing! I wonder if this would also be available for Woo experimentation?

Reply:
I believe they do it similarly, since they also rely on the Jetpack connection in Woo. I'm not sure if they use the same package tho.